plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -340,25 +333,21 @@ instance HasConstantIn DefaultUni term => ReadKnownIn DefaultUni term Int64 wher
             -- We don't make use here of `toIntegralSized` because of performance considerations,
             -- see: https://gitlab.haskell.org/ghc/ghc/-/issues/19641
             -- OPTIMIZE: benchmark an alternative `integerToIntMaybe`, modified from 'ghc-bignum'
-            if fromIntegral (minBound :: Int64) <= i && i <= fromIntegral (maxBound :: Int64)
-                then pure $ fromIntegral i
-                else throwing_ _EvaluationFailure
+#if WORD_SIZE_IN_BITS == 64
+            if fromIntegral (minBound :: Int) <= i && i <= fromIntegral (maxBound :: Int)
+            then pure $ fromIntegral i
+            else throwing_ _EvaluationFailure
+#else
+            -- hlint (or cpphs?) has problems parsing the #error cpp directive. Skip hlint then.
+#ifndef __HLINT__
+#error unsupported WORD_SIZE_IN_BITS. We only support 64-bit architectures.
+#endif
+#endif
     {-# INLINE readKnown #-}
 
 instance KnownTypeAst DefaultUni Int where
     toTypeAst _ = toTypeAst $ Proxy @Integer
 
--- See Note [Integral types as Integer].
```

Comment:
I was suggesting to just keep these instances, but put the whole instance declaration in the CPP, i.e.
```
#if WORD_SIZE_IN_BITS == 64
instance HasConstantIn DefaultUni term => MakeKnownIn DefaultUni term Int where ...
#endif 
```
That way we can keep the `Int64` instance free of conditional logic. I think that's kind of nice because AFAICT the `Int64` instance is correct regardless of the word size, which is sort of nice. We also don't need the `#error`, since it will simply mean that any other uses of `Int` in unlifting will fail to compile because the instance won't exist.

I'm partly just trying to think what expresses the truth of the situation as clearly as possible. I _think_ the main thing we want to avoid is casting from `Int` to `Int64` or vice versa in builtins unless they're identical. And having a conditionally available instance for `Int` kind of expresses that.

Anyway, I think this is fine.